Updates to Backup & Recovery section (#892)
BitcoinDesign/Guide: guide/daily-spending-wallet/backup-and-recovery/cloud-backup.md

-->

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/backing-up-a-wallet-cloud.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/backing-up-a-wallet-cloud@2x.png"
   mobile = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/backing-up-a-wallet-cloud-mobile.png"
   mobileRetina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/backing-up-a-wallet-cloud-mobile@2x.png"
   alt-text = "Backing up a recovery phrase chapter header image"
   width = 1600
   height = 600
   layout = "full-width"
%}

# Automatic cloud backup

## How it works

> The user’s wallet recovery data and channel states are backed up on their respective cloud storage provider. The [recovery phrase]({{ '/guide/glossary/#recovery-phrase' | relative_url }}) only needs to be backed up once. However, channel states need to be backed up frequently because they change every time this wallet sends or receives a payment; therefore, the channel state is backed up in the cloud automatically every time it is changed.

Reviewer: ```suggestion
This wallets recovery data and channel states are backed up on their respective cloud storage provider. The [recovery phrase]({{ '/guide/glossary/#recovery-phrase' | relative_url }}) only needs to be backed up once. However, channel states need to be backed up frequently because they change every time this wallet sends or receives a payment; therefore, the channel state is backed up in the cloud automatically every time it is changed.
```

Author: I don't like this rephrasing. It changes the meaning of the sentence.

> If the user’s device is lost, stolen, or broken, they can use the backup from their cloud storage account to regain access to their funds. This could include common ones like Apple iCloud and Google Drive, open source ones like NextCloud, or even a [Lightning Wallet Server]({{'/guide/how-it-works/lightning-services/#backups' | relative_url}}).  To learn more about the technical details, hop over to the [private key management]({{ '/guide/private-key-management/introduction' | relative_url }}) section on this topic.

Reviewer: ```suggestion
If the user’s phone is lost, stolen, or broken, they can use the backup from their cloud storage account to regain access to their funds. This could include common ones like Apple iCloud and Google Drive, open source ones like NextCloud, or even a [Lightning Wallet Server]({{'/guide/how-it-works/lightning-services/#backups' | relative_url}}).  To learn more about the technical details, hop over to the [private key management]({{ '/guide/private-key-management/introduction' | relative_url }}) section on this topic.
```

Third party backup solutions like a LWS should be encrypted somehow, especially if the recovery phrase is part of that backup.

This method provides users with the feeling that they can rely on their cloud provider to regain access to their funds rather than putting the responsibility entirely on themselves. It also significantly reduces friction during onboarding.

Below is what this wallet’s automatic cloud backup user interface looks like.

<div class="image-slide-gallery">

> {% include picture.html

Reviewer: This gallery is probably not the best place but we need content on what it would look like in the UI when a backup is made every time a payment is sent or received and what it would look like if that backup failed (maybe cloud provider is down?). Some kind of notification indicating the backup failed and some actions to fix it. 

In the settings -> backup & recovery it could also state "Last backup was made xx/xx/xxxx" for extra assurance.

Author: I don't think there should be an overt indication in the UI when a backup successfully happens. It's part of the normal expected functioning of the app -- the user doesn't need to be notified every time the app behaves as expected.

There could be some way to check this in the settings screen. Didn't we discuss doing a Settings section for this app? Maybe that would be a good section to show this?

Displaying an error state could be interesting, though.

Author: Added an error handling section.

Reviewer: Agree, can have a stub about latest back up in that settings page of which Christoph has opened an issue for

>    image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/cloud-backup-prompt.png"

Author: Done

%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/cloud-backup-PIN.png"

Reviewer 2: Keypad should not include alphabet characters.

Author: Done

</div>

## How to explain it

As mentioned in the previous section, this wallet prompts the user to backup during [First Use](https://bitcoin.design/guide/daily-spending-wallet/first-use/) and, if they opt-out, reminds them again once they have received funds.

The wallet explains to the user that their backup is stored on the cloud, and that they can regain access to their funds on another device using their respective cloud account and bitcoin wallet app. The app also explains that if somebody else gains access to your cloud account information, they can also gain access to your funds.

{% include tip/open.html color="green" icon="check" label="Do: Inform your Users" %}

Before creating a wallet backup via cloud backup, make sure the user knows what they are about to do. Once the backup is complete, provide a summary and some reassurances before moving on. Let them know where in their cloud storage they can find their backup, and give the file an understandable name. For example, a filename like `BitcoinWalletBackup_Samantha` conveys far more meaning than a randomly generated number.

{% include tip/close.html %}

## Tradeoffs

> By not making users manually backup, you decrease the risk of self-inflicted loss of funds. Automatic cloud backups put less pressure and responsibility on the user.

Reviewer: ```suggestion
By not making users [manually backup]({{ '/guide/daily-spending-wallet/backup-and-recovery/manual-backup/' | relative_url }}), you decrease the risk of self-inflicted loss of funds. Automatic cloud backups put less pressure and responsibility on the user.
```

While this is an easier way for users to set up a wallet, it brings in the potential threat of malicious third-party access. If a user’s cloud account is hacked, their funds could be compromised. This can be mitigated by encrypting the wallet recovery data before sending it to the cloud provider. However, the user could still lose access to funds if they forget their cloud login information or the cloud provider blocks their access.

> By bringing cloud storage providers into the equation, we assume that users trust their providers. This also raises the question of [decentralization]({{ 'guide/getting-started/principles/#decentralization' | relative_url }}), a guiding principle of the bitcoin space, and could be considered in conflict with the popular saying, “not your keys, not your coins.”

Reviewer: A possible remedy is use multiple backups, like one with the LSP encrypted, one on cloud provider

Author: Not a bad idea, but I think it's another level of complexity that's not necessary for this daily spending wallet

</div>

## How to explain it

> As mentioned in the previous section, this wallet prompts the user to backup during [First Use](https://bitcoin.design/guide/daily-spending-wallet/first-use/) and, if they opt-out, reminds them again once they have received funds.

Reviewer 2: ```suggestion
As mentioned in the previous section, this wallet prompts the user to back up during [First Use](https://bitcoin.design/guide/daily-spending-wallet/first-use/) and, if they opt-out, reminds them again once they have received funds.
```

{% include tip/open.html color="green" icon="check" label="Do: Inform your Users" %}

Before creating a wallet backup via cloud backup, make sure the user knows what they are about to do. Once the backup is complete, provide a summary and some reassurances before moving on. Let them know where in their cloud storage they can find their backup, and give the file an understandable name. For example, a filename like `BitcoinWalletBackup_Samantha` conveys far more meaning than a randomly generated number.

{% include tip/close.html %}

## Error Handling

<div class="center" markdown="1">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/cloud-backup-error.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/cloud-backup-error@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/automatic-cloud-backup/cloud-backup-error@2x.png"
   alt-text = "Screen showing an error message for a failed auto cloud backup"

>    caption = "This wallet notifies the user when there is a backup error and provides next steps"

Reviewer 2: ```suggestion
   caption = "This wallet notifies the user when there is a backup error and provides next steps."
```

   width = 250
   height = 541
   layout = "float-right-desktop -background -shadow"
%}

> In normal conditions, this wallet performs an automatic cloud backup after sending or receiving any payment. However, there could be situations where this fails to happen, such as an error or outage on the side of the cloud provider. Regardless of the reason, it's good to notify the user if there is a problem. This wallet offers a list of common issues and solutions, and also offers the ability to submit a support request.

Reviewer 2: Instead of a support request, maybe first send user to the settings of their cloud provider and ensure their account is still working as it should?

I'd also recommend sentence-casing button labels.

Author: Updated. I sentence-cased the buttons. For sending the user to their cloud provider, I think that would fit under "view common issues". I updated the text to drive people to that first, and then a support request if that doesn't work.

</div>

## Tradeoffs

> By not making users [manually backup]({{ '/guide/daily-spending-wallet/backup-and-recovery/manual-backup/' | relative_url }}), you decrease the risk of self-inflicted loss of funds. Automatic cloud backups put less pressure and responsibility on the user.

Reviewer 2: ```suggestion
By not making users [manually back up]({{ '/guide/daily-spending-wallet/backup-and-recovery/manual-backup/' | relative_url }}), you decrease the risk of self-inflicted loss of funds. Automatic cloud backups put less pressure and responsibility on the user.
```
